### ISSUE TITLE
Build fails when I have an SSR page and gatsby-source-contentful working together

### ISSUE THREAD

**Issue by fedehub93** [gatsbyjs/gatsby#37632]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

When I run the build with `gatsby build` I receive the **ERROR #98001 and Error: Worker exited before finishing task**.
I've noticed this problem using the gatsby-source-contentful-plugin with an SSR page.
As I remove the SSR page everything works fine.
Similarly disabling gatsby-source-contentful and leaving the SSR page everything works fine.

I also tested this with gatsby@next.

### Reproduction Link

https://github.com/fedehub93/gatsby-contentful-ssr-page

### Steps to Reproduce

1. `npm init gatsby`, I choose typescript configuration
2. `yarn add gatsby-source-contentful gatsby-plugin-image gatsby-plugin-sharp gatsby-transformer-sharp`
3. configure `gatsby-config.ts` with plugins and put your **spaceId** and **accessToken** in `gatsby-source-contentful` options:
```
  plugins: [
    {
      resolve: "gatsby-source-contentful",
      options: {
        spaceId: "<PUT_YOUR_SPACE_ID>",
        accessToken: "<PUT_YOUR_ACCESS_TOKEN>",
      },
    },
    `gatsby-plugin-image`,
    `gatsby-plugin-sharp`,
    `gatsby-transformer-sharp`,
  ],
```
5. add a new file in `src/pages/search.tsx` with following code:

```
import React from "react";
import { GetServerData, PageProps } from "gatsby";

type ServerDataProps = {
  title: string;
};

const Search = ({
  serverData,
}: PageProps<object, object, unknown, ServerDataProps>): JSX.Element => (
  <h1>{serverData.title}</h1>
);

export default Search;

export const getServerData: GetServerData<ServerDataProps> = async (props) => {
  return {
    props: {
      title: "Test SSR",
    },
  };
};

```
6. Run `gatsby build`

### Expected Result

The build runs successfully.

### Actual Result

I have this error:
```
failed Validating Rendering Engines - 1.409s

 ERROR #98001  ENGINE.VALIDATION

Built Rendering Engines failed validation failed validation.

Please open an issue with a reproduction at https://gatsby.dev/new-issue for more help.



  Error: Worker exited before finishing task

  - index.js:205 ChildProcess.<anonymous>
    [test-ssr]/[gatsby-worker]/dist/index.js:205:41
```

### Environment

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
  Binaries:
    Node: 18.0.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.6.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.7
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (109.0.1518.78)
  npmPackages:
    gatsby: ^5.6.0 => 5.6.0
    gatsby-plugin-image: ^3.6.0 => 3.6.0
    gatsby-plugin-sharp: ^5.6.0 => 5.6.0
    gatsby-source-contentful: ^8.6.0 => 8.6.0
    gatsby-source-filesystem: ^5.6.0 => 5.6.0
    gatsby-transformer-sharp: ^5.6.0 => 5.6.0
```

**Comment by LekoArts** [comment]
Hi!

Your issue seems to be similar to https://github.com/gatsbyjs/gatsby/issues/37640 - @tdelgio can you share what caused this in Contentful?

@fedehub93 we won't be able to help you with this reproduction at it seems specific to your Contentful instance. You'd need to create a space which credentials you could share freely.

**Comment by fedehub93** [comment]
> Hi!
> 
> Your issue seems to be similar to #37640 - @tdelgio can you share what caused this in Contentful?
> 
> @fedehub93 we won't be able to help you with this reproduction at it seems specific to your Contentful instance. You'd need to create a space which credentials you could share freely.

Hi, thank you for the quick response.

I created a new environment with a new space on contentful to be able to give you some test credentials to reproduce the error.

I have updated my repo in the reproduction link with these new credentials.

The error is also returned to me after creating a very simple content model on contentful with only one field which is a Rich Text Field in my case.

No records added.

**Comment by SebastianKoch27** [comment]
Are there any new information about this issue?

**Comment by tdelgio** [comment]
I downgrade Gatsby and plug-ins versions and it works.
It seems to be a compatibility problem between versions.

```

  "dependencies": {
    "gatsby": "^4.23.1",
    "gatsby-plugin-gatsby-cloud": "^5.6.0",
    "gatsby-plugin-image": "^2.23.1",
    "gatsby-plugin-manifest": "^5.6.0",
    "gatsby-plugin-offline": "^6.6.0",
    "gatsby-plugin-react-helmet": "^5.0.0",
    "gatsby-plugin-sharp": "^4.23.1",
    "gatsby-source-contentful": "^7.21.1",
    "gatsby-source-filesystem": "^5.6.0",
    "gatsby-transformer-sharp": "^4.23.1",
    "prop-types": "^15.7.2",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-helmet": "^6.1.0"
  },

```

**Comment by LekoArts** [comment]
We updated `babel` in https://github.com/gatsbyjs/gatsby/pull/37568 which contained this PR: https://github.com/babel/babel/pull/15290. It changes the syntax for dynamic imports which currently breaks the engine creation & validation.

For now, use `gatsby-source-contentful@8.3.0` until this is fixed.

**Comment by SebastianKoch27** [comment]
Sorry for this question, but what about testing before deployment. And why this took 6 days to find the issue?

**Comment by LekoArts** [comment]
A fix for this was published in `gatsby-source-contentful@8.6.1`
